From 44ad4341d07d623ba2bcbe0eaa20f7799d2013ea Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Tue, 17 Jun 2014 01:47:11 +0530
Subject: [PATCH 22/39] ASoC: soc-fw: KW fix for DAI

Issue: GMINL-3679

Fix KW ERROR:
Object 'p' was freed at line xxx after being freed by
calling 'kfree' at line xxx

Change-Id: Ie86bd80c298b9eeb66541cae6fd9c1f1674d301d
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/soc-fw.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index 2dece11..72f6b78 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -1644,11 +1644,12 @@ void snd_soc_fw_dcontrols_remove_codec(struct snd_soc_codec *codec,
 	struct soc_enum *se, *next_se;
 	struct soc_bytes_ext *sb, *next_sb;
 	struct snd_card *card = codec->card->snd_card;
-	const unsigned int *p = NULL;
+	const unsigned int *p;
 	int i;
 
 	list_for_each_entry_safe(sm, next_sm, &codec->dmixers, list) {
 
+		p = NULL;
 		if (sm->index != index)
 			continue;
 
@@ -1694,11 +1695,13 @@ void snd_soc_fw_dcontrols_remove_platform(struct snd_soc_platform *platform,
 	struct soc_enum *se, *next_se;
 	struct soc_bytes_ext *sb, *next_sb;
 	struct snd_card *card = platform->card->snd_card;
-	const unsigned int *p = NULL;
+	const unsigned int *p;
 	int i;
 
 	list_for_each_entry_safe(sm, next_sm, &platform->dmixers, list) {
 
+		p = NULL;
+
 		if (sm->index != index)
 			continue;
 
@@ -1744,11 +1747,12 @@ void snd_soc_fw_dcontrols_remove_card(struct snd_soc_card *soc_card,
 	struct soc_enum *se, *next_se;
 	struct soc_bytes_ext *sb, *next_sb;
 	struct snd_card *card = soc_card->snd_card;
-	const unsigned int *p = NULL;
+	const unsigned int *p;
 	int i;
 
 	list_for_each_entry_safe(sm, next_sm, &soc_card->dmixers, list) {
 
+		p = NULL;
 		if (sm->index != index)
 			continue;
 
-- 
1.9.1

